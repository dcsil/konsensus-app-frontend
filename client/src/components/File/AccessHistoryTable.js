import { Table, Card, CardHeader } from 'reactstrap';
// core components
import ActionRow from './ActionRow';

const AccessHistoryTable = ({ actions, user }) => {
  console.log(user);
  return (
    <Card className="bg-default shadow">
      <CardHeader className="bg-transparent border-0">
        <h3 className="text-secondary mb-0">Access History</h3>
      </CardHeader>
      <Table
        className="align-items-center table-dark table-flush pt-3"
        responsive
      >
        <thead className="thead-dark">
          <tr>
            <th scope="col" />
            <th scope="col">Name</th>
            <th scope="col">Role</th>
            <th scope="col">Action</th>
            <th scope="col">Status</th>
            <th scope="col">Time</th>
            <th scope="col" />
          </tr>
        </thead>
        <tbody>
          <ActionRow
            firstName={user ? user.firstName : ''}
            lastName={user ? user.lastName : ''}
            role={user ? user.role : ''}
            action={'updated'}
            time={actions[0].updatedAt}
          />
          <ActionRow
            firstName={user ? user.firstName : ''}
            lastName={user ? user.lastName : ''}
            role={user ? user.role : ''}
            action={'created'}
            time={actions[0].createdAt}
          />
        </tbody>
      </Table>
    </Card>
  );
};

export default AccessHistoryTable;
