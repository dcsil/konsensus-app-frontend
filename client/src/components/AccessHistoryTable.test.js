import React from 'react';
import { shallow } from 'enzyme';
import AccessHistoryTable from './File/AccessHistoryTable';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Test Sidebar component', () => {
  const wrapper = shallow(<AccessHistoryTable />);
  expect(wrapper).toMatchSnapshot();
});
