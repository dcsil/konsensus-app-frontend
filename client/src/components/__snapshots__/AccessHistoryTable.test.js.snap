// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AccessHistoryTable component 1`] = `ShallowWrapper {}`;
